tests/nodal_properties_test.cc

```diff
@@ -0,0 +1,166 @@
+#include <Eigen/Dense>
+#include <memory>
+
+#include "catch.hpp"
+
+#include "nodal_properties.h"
+
+//! \brief Check NodalProperties struct
+TEST_CASE("NodalProperties is checked", "[nodal_properties]") {
+  // Number of nodes
+  const unsigned nnodes = 3;
+  // Number of materials
+  const unsigned nmaterials = 2;
+  // Tolerance
+  const double tolerance = 1.E-7;
+
+  // Check scalar property creation
+  SECTION("Create scalar properties") {
+    // Declare nodal properties
+    mpm::NodalProperties nodal_properties;
+
+    // Define properties to be created
+    std::string property1 = "masses";
+    std::string property2 = "areas";
+
+    // Define dimension (1D for scalar)
+    const unsigned dim = 1;
+
+    // Check property creation
+    REQUIRE_NOTHROW(
+        nodal_properties.create_property(property1, nnodes * dim, nmaterials));
+    REQUIRE_NOTHROW(
+        nodal_properties.create_property(property2, nnodes * dim, nmaterials));
+
+    // Check size of matrix of property data
+    REQUIRE(nodal_properties.properties_.at(property1).rows() == 3);
+    REQUIRE(nodal_properties.properties_.at(property1).cols() == 2);
+    REQUIRE(nodal_properties.properties_.at(property1).size() == 6);
+    REQUIRE(nodal_properties.properties_.at(property2).rows() == 3);
+    REQUIRE(nodal_properties.properties_.at(property2).cols() == 2);
+    REQUIRE(nodal_properties.properties_.at(property2).size() == 6);
+
+    // Check if values in nodal properties are equal to 0.0
+    for (int i = 0; i < nnodes; ++i) {
+      for (int j = 0; j < nmaterials; ++j) {
+        for (int k = 0; k < dim; ++k) {
+          REQUIRE(nodal_properties.property(property1, i, j, dim)(k, 0) ==
+                  Approx(0.0).epsilon(tolerance));
+          REQUIRE(nodal_properties.property(property2, i, j, dim)(k, 0) ==
+                  Approx(0.0).epsilon(tolerance));
+        }
+      }
+    }
+
+    // clang-format off
+    Eigen::Matrix<double,3,2> data1;
+    data1 << 4.2, 3.1,
+             0.5, 1.7,
+             6.6, 8.2;
+    Eigen::Matrix<double,3,2> data2;
+    data2 << 4.1, 3.2,
+             6.8, 5.1,
+             9.1, 2.0;
+    // clang-format on
+
+    // Update values in the property data matrix and check update
+    for (int i = 0; i < nnodes; ++i) {
+      for (int j = 0; j < nmaterials; ++j) {
+        REQUIRE_NOTHROW(nodal_properties.assign_property(
+            property1, i, j, data1.block<dim, 1>(i * dim, j), dim));
+        REQUIRE_NOTHROW(nodal_properties.assign_property(
+            property2, i, j, data2.block<dim, 1>(i * dim, j), dim));
+      }
+    }
+
+    // Check values in matrix of property data
+    for (int i = 0; i < nnodes; ++i) {
```

Comment:
What will be the default value of inactive nodes? What will happen if one node was active and become inactive as particle moves?

Reply:
This is just for testing. We have `status` parameter to deal with those situations

Reply:
I see. So for each node the array size will just be dim * nmaterials for each property?

Reply:
There will be one property pool, if you add your node to it, it means it's active and it will get a handle, if it is a non-active node, you won't get a handle or add to pool

Reply:
Does that mean the size of the pool change with the number of active nodes? It means it requires you either to construct the pool at every step (size nodal quantity is reset anyway) or to resize the pool.

Reply:
Yes, the pool size will change dynamically, but only when load rebalancing happens, not at every step

Reply:
Do you consider special cases, such as when we have 2 materials at a shared node? I am thinking about contact or two-phase two-layers implementation

Reply:
That's the whole idea behind this property pool to handle contact

Reply:
Awesome!